.circleci/config.yml

```diff
@@ -1345,32 +1345,44 @@ workflows:
             value: << pipeline.git.branch >>
     jobs:
       - release-checks
-      - hold:
+      - hold-for-tag:
           type: approval
           requires:
             - release-checks
       - tag-release-branch:
           requires:
-            - hold
+            - hold-for-tag
+
+  deploy-tag:
+    when:
+      and:
+        - equal: [push, << pipeline.trigger_source >>]
+        - exists: << pipeline.git.tag >>
+        - not:
+          matches:
+            pattern: ".*-SNAPSHOT$"
+            value: << pipeline.git.tag >>
+    jobs:
+      - hold-for-release:
+          type: approval
```

Comment:
is having a second hold here too much?

Reply:
Kinda feel like we have to have _something_, otherwise any accidental push of a tag in a branch that has a PR triggers a release